frontend/src/components/lists/data.jsx

```diff
@@ -21,6 +27,49 @@ export default class ListsData extends React.Component {
         this.props.onEdit(item.id);
     }
 
+    onDragStart(e, item) {
+        // Hide default thumbnail
+        e.dataTransfer.setDragImage(new Image(), 0, 0);
+
+        this.setState({
+            draggingItem: item,
+            draggingData: this.props.data
+        });
+    }
+
+    onDragEnter(e, draggedOverId) {
+        // Required so that onDrop fires (https://www.quirksmode.org/blog/archives/2009/09/the_html5_drag.html)
+        e.preventDefault();
+        e.dataTransfer.dropEffect = 'copyMove';
+
+        const { draggingItem } = this.state;
+
+        if(draggingItem.id !== draggedOverId) {
+            const draggingData = this.state.draggingData.filter(item => item.id !== draggingItem.id);
+
+            const dropId = this.state.draggingData.findIndex(item => item.id === draggedOverId);
+            draggingData.splice(dropId, 0, this.state.draggingItem);
+
+            this.setState({ draggingData });
+        }
+    }
+
+    onDragOver(e) {
+        // Required so that onDrop fires (https://www.quirksmode.org/blog/archives/2009/09/the_html5_drag.html)
+        e.preventDefault();
+        e.dataTransfer.dropEffect = 'copyMove';
+    }
+
+    onDrop() {
+        const { draggingItem } = this.state;
+
+        const pos = this.props.data.findIndex(item => item.id === draggingItem.id);
+        const newPos = this.state.draggingData.findIndex(item => item.id === draggingItem.id);
```

Comment:
Just to check, is `draggingItem` a different reference to the DOM element `item`? Otherwise I think we can use `indexOf` here.

Reply:
Ah good spot, they're not even DOM elements but objects from the props so `indexOf` will work